src/hb-serialize.hh

```diff
@@ -127,12 +129,13 @@ struct hb_serialize_context_t
     object_pool.fini ();
   }
 
-  bool in_error () const { return errors & HB_SERIALIZE_ERR_OTHER; }
+  bool in_error () const { return bool (errors); }
 
-  bool successful () const { return !in_error (); }
+  bool successful () const { return !bool (errors); }
 
-  bool ran_out_of_room () const { return errors & HB_SERIALIZE_ERR_OUT_OF_ROOM; }
-  bool offset_overflow () const { return errors & HB_SERIALIZE_ERR_OFFSET_OVERFLOW; }
+  bool ran_out_of_room () const { return errors & HB_SERIALIZE_ERROR_OUT_OF_ROOM; }
+  bool offset_overflow () const { return errors & HB_SERIALIZE_ERROR_OFFSET_OVERFLOW; }
+  bool only_offset_overflow () const { return errors == HB_SERIALIZE_ERROR_OFFSET_OVERFLOW; }
```

Comment:
Please add `HB_NODISCARD` so we'll notice if someone calls them thinking that they mark error. Or maybe we should remove / rename completely and deal with errors directly. WDUT?

Reply:
I think just the NO_DISCARD should be fine. Will do that for now.